test/functional/plugin/lsp_spec.lua

```diff
@@ -813,6 +813,74 @@ describe('LSP', function()
     end)
   end)
 
+  describe('apply_text_document_edit', function()
+    before_each(function()
+      insert(dedent([[
+        First line of text
+        Second line of text]]))
+    end)
+    it('correctly goes ahead with the edit when all is normal', function()
+      local text_document_edit = {
+        edits = {
+          range = {
+            start = { line = 1, character = 1 };
+            ["end"] = { line = 1, character = 2 };
+          },
+          newText = "bar"
+        },
+        textDocument = {
+          uri = "/fake/uri";
+          version = 5
+        }
+      }
+      exec_lua('vim.lsp.util.apply_text_document_edit(...)', text_document_edit, 1)
+      eq({
+        'barFirst line of text';
+        'Second line of text';
+      }, buf_lines(1))
+    end)
+    it('correctly goes ahead with the edit whe the version is nil', function()
+      local text_document_edit = {
+        edits = {
+          range = {
+            start = { line = 1, character = 1 };
+            ["end"] = { line = 1, character = 2 };
+          },
+          newText = "bar"
+        },
+        textDocument = {
+          uri = "/fake/uri";
+          version = vim.NIL
```

Comment:
I also think this is wrong. I need to check against this in `util.lua`, and locally this is what the version is if it comes in as `null`, but this fails for a different reason in the test.

```
lsp_spec.lua:853: attempt to index global 'vim' (a nil value)
```
What should I be setting this as?

This is my first stab at Lua, so I'm sure these are probably obvious...

Reply:
If we use vim table, we should execute Lua on Nvim, not normal Lua runtime.
In other words, you can use vim table in `exec_lua` function.